controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -386,16 +498,14 @@ public String getScopeName() {
 
     @Override
     public CompletableFuture<TxnStatus> checkTransactionStatus(final UUID txId) {
-        CompletableFuture<Integer> epochFuture = getTransactionEpoch(txId).handle((epoch, ex) -> {
-            if (ex != null && ExceptionHelpers.getRealException(ex) instanceof DataNotFoundException) {
-                return null;
-            } else if (ex != null) {
-                throw new CompletionException(ex);
-            }
-            return epoch;
-        });
-
-        return verifyLegalState(epochFuture.thenCompose(x -> {
+        return verifyLegalState(() -> getTransactionEpoch(txId).handle((epoch, ex) -> {
```

Comment:
Why is verifyLegalState needed for read operation like `checkTransactionStatus`?